apiserver/facades/agent/uniter/newlxdprofile.go

```diff
@@ -211,12 +217,23 @@ func (u *LXDProfileAPIv2) LXDProfileName(args params.Entities) (params.StringRes
 			result.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrPerm)
 			continue
 		}
-		unit, machine, err := u.getLXDProfileUnitMachineV2(tag)
+		unit, _, err := u.getLXDProfileUnitMachineV2(tag)
 		if err != nil {
 			result.Results[i].Error = apiservererrors.ServerError(err)
 			continue
 		}
-		name, err := u.getOneLXDProfileName(unit, machine)
+
+		machineTagID, err := unit.AssignedMachineId()
```

Comment:
This and step below should ultimately be a single call - get the machine UUID for a unit, but I'm aware that this linkage does not properly reside in Dqlite at present.

Reply:
Indeed, I didn't add a TODO because I have to go over these methods anyway once we finish units and apps to rewrite this watcher.

Reply:
question: is there a card to rewrite this watcher?